components/health_controller/src/reporters.rs

```diff
@@ -109,34 +195,42 @@ impl RaftstoreReporter {
         }
     }
 
-    pub fn tick(&mut self, store_maybe_busy: bool) -> SlowScoreTickResult {
+    pub fn tick(&mut self, store_maybe_busy: bool, factor: InspectFactor) -> SlowScoreTickResult {
         // Record a fairly great value when timeout
         self.slow_trend.slow_cause.record(500_000, Instant::now());
 
+        // healthy: The health status of the current store.
+        // last_tick_finished: The last tick of all factors is finished.
```

Comment:
Nope, the reporter can set the state to be `healthy` only if the last ticks on different factors are finished as expected.

One scenario:
- Currently, not healthy => `healthy == false`.
- T0 => SlowScore on `raft` has finished its last tick, then it will set the state of the node to be `healthy == true`.
- T1 => SlowScore on `kvdb` has not finished its last tick, then it check the `tick()`, and if the `tick % round != 0`, it will return (None, false)`.

Then, SlowScore on `kvdb` won't change the state to be `healthy == false` as expected.

Reply:
OK, got it. We need all factors to set the health state properly. Maybe `all_ticks_finished` is a better name now.